src/components/tabs/tabs.js

```diff
@@ -292,65 +306,58 @@ export default Vue.extend({
   created() {
     let tabIdx = parseInt(this.value, 10)
     this.currentTab = isNaN(tabIdx) ? -1 : tabIdx
-    // Create private non-reactive prop
-    this._bvObserver = null
     // For SSR and to make sure only a single tab is shown on mount
     // We wrap this in a `$nextTick()` to ensure the child tabs have been created
     this.$nextTick(() => {
       this.updateTabs()
     })
   },
   mounted() {
+    // Call `updateTabs()` just in case...
+    this.updateTabs()
     this.$nextTick(() => {
-      // Call `updateTabs()` just in case...
-      this.updateTabs()
-      // Observe child changes so we can update list of tabs
-      this.setObserver(true)
-      // Flag we are now mounted and to switch to DOM for tab probing
+      // Flag we are now mounted and to switch to DOM for tab probing.
+      // As this.$slots.default appears to lie about component instances
+      // after b-tabs is destroyed and re-instantiated.
+      // And this.$children does not respect DOM order.
       this.isMounted = true
     })
   },
   deactivated() /* istanbul ignore next */ {
-    this.setObserver(false)
     this.isMounted = false
   },
   activated() /* istanbul ignore next */ {
     let tabIdx = parseInt(this.value, 10)
     this.currentTab = isNaN(tabIdx) ? -1 : tabIdx
     this.$nextTick(() => {
       this.updateTabs()
-      this.setObserver(true)
       this.isMounted = true
     })
   },
-  beforeDestroy() /* istanbul ignore next */ {
-    this.setObserver(false)
+  destroyed() {
+    // Ensure no references to child instances exist
+    this.tabs = []
   },
   methods: {
-    setObserver(on) {
-      if (on) {
-        // Make sure no existing observer running
-        this.setObserver(false)
-        // Watch for changes to <b-tab> sub components
-        this._bvObserver = observeDom(this.$refs.tabsContainer, this.updateTabs.bind(this), {
-          childList: true,
-          subtree: false,
-          attributes: true,
-          attributeFilter: ['style', 'class']
+    registerTab(tab) {
+      if (!arrayIncludes(this.registeredTabs)) {
```

Comment:
`if (!arrayIncludes(this.registeredTabs, tab)) {`

Reply:
Oops

Reply:
good catch